File: Algorithms/Arrays/Kadane/kadane_algo.cpp
#include<iostream>
using namespace std;

void kadane_algo(int arr[], int n) {

    int csum = arr[0];
    int osum = arr[0];

    for (int i = 1; i < n; i++) {
        if (csum < 0) {
            csum = arr[i];
        } else {
            csum += arr[i];
        }

        if (csum > osum) {
            osum = csum;
        }
    }

    cout << "Max sum is " << osum << endl;
}

int main() {

    int n;
    cout << "Enter the size of the array: ";
    cin >> n;
    cout << "\nEnter the elements of the array" << endl;
    int arr[n];

    for (int i = 0; i < n; i++)
        cin >> arr[i];

    kadane_algo(arr, n);
    return 0;
}

File: Algorithms/Arrays/Kadane/kadane_algo.py
# Function to find the maximum subarray sum
# using Kadane's Algorithm
def kadane_algo(arr):

    csum = osum = arr[0]
    for i in range(1, len(arr)):

        if csum < 0:
            csum = arr[i]
        else:
            csum += arr[i]

        if csum > osum:
            osum = csum

    print("Max sum is", osum)

def main():

    n = int(input("Enter the size of the array: "))
    print("Enter the elements of the array")
    arr = [None] * n

    for i in range(n):
        arr[i] = int(input())

    kadane_algo(arr)

if __name__ == "__main__":
    main()
